.github/workflows/build-and-run-node-test.yaml

```
@@ -3,11 +3,31 @@ name: "Build and test"
 on: ["push"]
 
 jobs:
- node-test:
+  check-paths:
```

nit: I think this'll make it a bit easier to understand what the check is going from the other jobs

```suggestion
  code-paths-changed:
```

E.g, we now do: "needs.code-paths-changed.outputs.run_tests == true"